inst/tutorials/r_and_rstudio_basic/r_and_rstudio_basic.Rmd

```diff
@@ -0,0 +1,284 @@
+
+```{r}
+2 + 2 
+```
+
+To declare a variable, follow the pattern of: `variable <- value`. Let's declare a variable `total` as the sum of two numbers.
+
+```{r}
+total <- 2 + 2
+```
+
+We access the value of `total`:
+
+```{r}
+total
+```
+
+We can use the value stored in `total`:
+
+```{r}
+total - 1
+```
+
+After declaring a variable, we can perform operations to change the value stored in the variable:
+
+```{r}
+total <- total - 1
+```
+
+Now `total` stores a new value:
+
+```{r}
+total
+```
+
+
+
+## Functions
+
+Functions are one of the basic units in programming. Generally speaking, a function takes some input and generates some output. Every R function follows the same basic syntax, where `function()` is the name of the function and `arguments` are the different parameters you can specify (i.e. your input):
+
+`function(argument1 = ..., argument2 = ..., ...)`
+
+You can treat functions as a black box and do not necessarily need to know how it works under the hood as long as your provided input conforms to a specific shape.
+
+![](/images/function.png){width=75%}
+
+For example, the function `sum()` expects numbers:
+
+```{r}
+sum(3, 5, 9, 18)
+```
+
+If you instead pass text as arguments to `sum()` you will receive an error:
+
+```{r error = TRUE}
+sum("Sum", "does", "not", "accept", "text!")
+```
+
+
+## R packages
+
+The first functions we will look at are used to install and load R packages. R packages are units of shareable code, containing functions that facilitate and enhance analyses. In simpler terms, think of R packages as iPhone Applications. Each App has specific capabilities that can be accessed when we install and then open the application. The same holds true for R packages. To use the functions contained in a specific R package, we first need to install the package, then each time we want to use the package we need to "open" the package by loading it.
+
+### Installing Packages
+
+In this tutorial, we will be using the "tidyverse" package. This package contains a versatile set of functions designed for easy manipulation of data.
+
+You should have already installed the "tidyverse" package using RStudio's graphical interface. Packages can also be installed by entering the function `install.packages()` in the console (to install a different package just replace "tidyverse" with the name of the desired package):
+
+```{r eval = FALSE}
+install.packages("tidyverse")
+```
+
+
+
+### Loading packages
+
+After installing a package, and *everytime* you open a new RStudio session, you need to first load (open) the packages you want to use with the `library()` function. This tells R to access the package's functions and prevents RStudio from lags that would occur if it automatically loaded every downloaded package every time you opened it.
+
+Packages can be loaded like this:
+
+```{r eval = FALSE}
+library(tidyverse)
+```
+
```

Comment:
It is a little tricker to load Bioconductor packages, as they are often not stored on the Comprehensive R Archive Network (CRAN) where most packages live. There is a package, however, that lives on CRAN and serves as an interface between CRAN and the Bioconductor packages.

To load a Bioconductor package, you must first install and load the BiocManager package, like so.

`install.packages("BiocManager")`
`library("BiocManager`)

You can then use the function `BiocManager::install()` to install a package. To install the `Annotate` package, we would execute the following code.

`BiocManager::install("annotate")`